CMakeLists.txt

```diff
@@ -38,17 +38,20 @@ message(STATUS)
 add_compile_options(-fPIC -Wall -Werror -Wextra -Wpedantic)
 set(CMAKE_CXX_STANDARD 17)
 
+if(WITH_TEST)
+    add_compile_definitions(AOS_CONFIG_TYPES_URL_LEN=512)
+    add_compile_definitions(AOS_CONFIG_FS_FILE_PATH_LEN=512)
+endif()
+
+set(AOS_CORE_LIB_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/external/aos_core_lib_cpp)
+set(AOS_CORE_COMMON_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
Do you need this var outside this project?

Reply:
Yes, this is necessary because we need to have paths to directories relative to the root of the current project in other CMake files. Since we include submodules not where the main CMake files are, but in the respective directories, I couldn't find a CMake variable that would reliably refer to the root directory. This is because the project that includes the submodule changes these CMake variable values. Therefore, a new variable was introduced to represent the project's root, and it needs to be adjusted in two cases: when we build the project separately or when it is used as a submodule.